accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,511 @@
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class AddedInOSPlatformVersionAttribute  : OSPlatformVersionAttribute
+    {
+        public AddedInOSPlatformVersionAttribute(string osPlatform,
+                                                 int major,
+                                                 int minor,
+                                                 int build,
+                                                 int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class RemovedInOSPlatformVersionAttribute  : OSPlatformVersionAttribute
+    {
+        public RemovedInOSPlatformVersionAttribute(string osPlatform,
+                                                   int major,
+                                                   int minor,
+                                                   int build,
+                                                   int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class ObsoletedInOSPlatformVersionAttribute  : OSPlatformVersionAttribute
+    {
+        public ObsoletedInOSPlatformVersionAttribute(string osPlatform,
+                                              int major,
+                                              int minor,
+                                              int build,
+                                              int revision);
+        public string Url { get; set; }
+    }
+}
+```
+
+### Platform context
+
+To determine the platform context of the call site the analyzer must consider
+the following two configurations:
+
+1. **Call-site**. Application of `AddedInOSPlatformVersionAttribute` to the containing
+   member, type, module, or assembly.
+
+2. **Project** The project TFMs and `TargetPlatformMinVersion`. If the project
+   has an platform-specific TFM (such as `net5.0-ios13.0`) we consider it to be
+   on the specified platform. If `TargetPlatformMinVersion` is set, we assume
+   that version, otherwise we assume the version as specified by the TFM.
+
+If (2) is platform neutral (e.g. `net5.0`), the platform context is entirely
```

Comment:
Is this apply same for `netcoreappX.X` and `netstandardX.X`? When TFM is `net4XX` or `uap` it count as windows only?